tests/algorithms/rational.py

```diff
@@ -104,6 +130,14 @@ def mat_shift(dst, src, s):
 g.message(f"Test zolotarev 1/sqrt(mat) approximation: {eps2}")
 assert eps2 < 1e-12
 
+# test zolotarev_sign and msfom with LR deflation
+signz = rat.zolotarev_sign(
+    {"eps": 1e-5, "inverter": msfom, "low": 0.4, "high": 3.4}
+)
+ssignz = signz(mat, msq_evals=evals[0:10], msq_evec=evec[0:10])
```

Comment:
Taking evals and evec in here is also against gpt design principles.  Look at how we do deflation in general solvers in a modular manner with sequence inverters.